ext/har/logger.go

```diff
@@ -0,0 +1,110 @@
+package har
+
+
+import (
+    "encoding/json"
+    "net/http"
+    "os"
+    "sync"
+    "time"
+
+    "github.com/elazarl/goproxy"
+)
+
+// Logger implements a HAR logging extension for goproxy
+type Logger struct {
+	mu             sync.Mutex
+	har            *Har
+	captureContent bool
+}
+
+// NewLogger creates a new HAR logger instance
+func NewLogger() *Logger {
+	return &Logger{
+		har: New(),
+	}
+}
+
+// OnRequest handles incoming HTTP requests
+func (l *Logger) OnRequest(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	// Store the start time in context for later use
+	if ctx != nil {
+		ctx.UserData = time.Now()
+	}
+	return req, nil
+}
+
+// OnResponse handles HTTP responses
+func (l *Logger) OnResponse(resp *http.Response, ctx *goproxy.ProxyCtx) *http.Response {
+	if resp == nil || ctx == nil || ctx.Req == nil || ctx.UserData == nil {
+		return resp
+	}
+
+	startTime, ok := ctx.UserData.(time.Time)
+	if !ok {
+		return resp
+	}
+
+	// Create HAR entry
+	entry := Entry{
+		StartedDateTime: startTime,
+		Time:           time.Since(startTime).Milliseconds(),
+		Request:        ParseRequest(ctx.Req, l.captureContent),
+		Response:       ParseResponse(resp, l.captureContent),
+		Cache:          Cache{},
```

Comment:
Instead of setting an empty object, just skip the struct value and the zero value will remain by default